internal/provider/resource_gitlab_group_test.go

```diff
@@ -199,27 +201,26 @@ func TestAccGitlabGroup_nested(t *testing.T) {
 					}),
 				),
 			},
-			// TODO In EE version, re-creating on the same path where a previous group was soft-deleted doesn't work.
-			// {
-			// 	Config: testAccGitlabNestedGroupConfig(rInt),
-			// 	Check: resource.ComposeTestCheckFunc(
-			// 		testAccCheckGitlabGroupExists("gitlab_group.foo", &group),
-			// 		testAccCheckGitlabGroupExists("gitlab_group.foo2", &group2),
-			// 		testAccCheckGitlabGroupExists("gitlab_group.nested_foo", &nestedGroup),
-			// 		testAccCheckGitlabGroupAttributes(&nestedGroup, &testAccGitlabGroupExpectedAttributes{
-			// 			Name:        fmt.Sprintf("nfoo-name-%d", rInt),
-			// 			Path:        fmt.Sprintf("nfoo-path-%d", rInt),
-			// 			Description: "Terraform acceptance tests",
-			// 			LFSEnabled:  true,
-			//			Visibility:            "public",     // default value
-			//			ProjectCreationLevel:  "maintainer", // default value
-			//			SubGroupCreationLevel: "owner",      // default value
-			//			TwoFactorGracePeriod:  48,           // default value
-			//			DefaultBranchProtection: 2,          // default value
-			// 			Parent:      &group,
-			// 		}),
-			// 	),
-			// },
+			{
+				Config: testAccGitlabNestedGroupConfig(rInt),
```

Comment:
A few ideas to make this test step more transparent:

* doesn't the group id stay the same after the transfer? We may be able to check for this instead of the non-intuitive `auto_devops_enabled` attribute check
* another idea could actually be to verify what you wanted to achieve (or at least what the PR title states): testing if underlying projects are preserved during a group transfer. Since these are acceptance tests I think it's okay to even introduce a separate test case to test that particular behavior.

WDYT?

Reply:
Checking the group id was my first idea. But this means that I have to keep a state between the test steps. This seemed a bit ugly to me. I will try this approach again though. :+1:

Reply:
I don't think it's necessarily a bad idea. Here is an example where it's already done: 
https://github.com/gitlabhq/terraform-provider-gitlab/blob/145117ca03d4f91a5fd2fec325c8a0c60106913e/internal/provider/resource_gitlab_project_issue_test.go#L92

Reply:
I have also tried the idea with the projects. But I don't think that works with the setup of the test steps. If the project is created in the Config with Terraform, then it is also created again after a re-create of the group. So I need to test if the project was recreated or not. So I am facing the same problem as with the groups. :man_juggling:

Reply:
You can create test projects outside of the terraform config, e.g.:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/145117ca03d4f91a5fd2fec325c8a0c60106913e/internal/provider/resource_gitlab_project_tag_test.go#L19

Reply:
... ah I see, then you are not able to easily attach it to that group under test 🤔

Reply:
I've added a test, that ensures the group id does not change